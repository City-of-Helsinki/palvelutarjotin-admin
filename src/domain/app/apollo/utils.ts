import apolloClient from './apolloClient';

export const clearApolloCache = async () => {
  apolloClient.stop();
  await Promise.all([apolloClient.resetStore(), apolloClient.clearStore()]);
};
