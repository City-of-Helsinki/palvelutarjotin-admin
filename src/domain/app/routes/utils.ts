import { ROUTES } from './constants';

export const getCmsPath = (slug?: string | null) => {
  return ROUTES.CMS_PAGE.replace('/:slug', slug ?? '');
};
