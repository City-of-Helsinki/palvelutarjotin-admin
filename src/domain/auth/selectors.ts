import { StoreState } from '../../types';

export const userAccessTokenSelector = (state: StoreState) =>
  state.authentication.tunnistamo.user?.access_token;

export const userSelector = (state: StoreState) =>
  state.authentication.tunnistamo.user;

export const apiTokenSelector = (state: StoreState) =>
  state.authentication.token.apiToken;

export const isLoadingUserSelector = (state: StoreState) =>
  state.authentication.tunnistamo.isLoadingUser ||
  state.authentication.token.isFetchingToken;

export const isAuthenticatedSelector = (state: StoreState) => {
  return (
    !!state.authentication.tunnistamo.user &&
    !!state.authentication.token.apiToken
  );
};
