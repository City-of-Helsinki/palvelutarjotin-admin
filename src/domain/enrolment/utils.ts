import { EnrolmentFieldsFragment } from '../../generated/graphql';

export const getEnrolmentFields = (
  enrolment: EnrolmentFieldsFragment | undefined | null
) =>
  enrolment
    ? {
        extraNeeds: enrolment.studyGroup.extraNeeds,
        language:
          enrolment.person?.language || enrolment.studyGroup.person.language,
        personInfo: [
          enrolment.person?.phoneNumber,
          enrolment.person?.emailAddress,
        ].filter((e) => e),
        studyGroupPersonInfo: [
          enrolment.studyGroup.person.phoneNumber,
          enrolment.studyGroup.person.emailAddress,
        ].filter((e) => e),
      }
    : {};
