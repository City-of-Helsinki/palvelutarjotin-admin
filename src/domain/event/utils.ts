import { SUPPORT_LANGUAGES } from '../../constants';
import { EventQuery } from '../../generated/graphql';
import { Language } from '../../types';
import { EVENT_PLACEHOLDER_IMAGES } from './constants';
/**
 * Get event placeholder image url
 * @param {string} id
 * @return {string}
 */
export const getEventPlaceholderImage = (id: string): string => {
  const numbers = id.match(/\d+/g);
  const sum = numbers
    ? numbers.reduce((prev: number, cur: string) => prev + Number(cur), 0)
    : 0;
  const index = sum % 4;

  return EVENT_PLACEHOLDER_IMAGES[index];
};

/**
 * Get first availanle language of an event
 * @param {object} eventData
 * @return {string}
 */
export const getFirstAvailableLanguage = (eventData: EventQuery): Language => {
  if (eventData.event?.name.en) return SUPPORT_LANGUAGES.EN;
  if (eventData.event?.name.fi) return SUPPORT_LANGUAGES.FI;
  if (eventData.event?.name.sv) return SUPPORT_LANGUAGES.SV;

  return SUPPORT_LANGUAGES.FI;
};
