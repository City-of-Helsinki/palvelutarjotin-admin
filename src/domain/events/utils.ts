import { isFuture } from 'date-fns';
import forEach from 'lodash/forEach';

import { EventFieldsFragment } from '../../generated/graphql';

export const hasOccurrences = (event: EventFieldsFragment) => {
  return Boolean(event.pEvent?.occurrences.edges.length);
};

export const hasComingOccurrences = (event: EventFieldsFragment) => {
  let hasComingItems = false;

  forEach(event.pEvent?.occurrences.edges, (edge) => {
    if (edge?.node?.startTime && isFuture(new Date(edge?.node?.startTime))) {
      hasComingItems = true;
      return false;
    }
  });

  return Boolean(hasComingItems);
};
