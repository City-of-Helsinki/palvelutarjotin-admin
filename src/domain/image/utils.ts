import { ImageDocument, ImageQuery } from '../../generated/graphql';
import { initializeApolloClient } from '../app/apollo/apolloClient';

export const getImageName = (id: string) => {
  const apolloClient = initializeApolloClient();
  const data = apolloClient.readQuery<ImageQuery>({
    query: ImageDocument,
    variables: { id },
  });
  return data?.image?.name;
};
