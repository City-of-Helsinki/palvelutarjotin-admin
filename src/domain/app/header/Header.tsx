import {
  Button as HDSButton,
  Header as HDSHeader,
  IconAngleRight,
  IconLinkExternal,
  IconSignout,
  IconUser,
  useOidcClient,
} from 'hds-react';
import * as React from 'react';
import {
  Language as RHHCLanguage,
  LanguageCodeEnum,
  MenuItem,
  Navigation,
} from 'react-helsinki-headless-cms';
import {
  useLanguagesQuery,
  useMenuQuery,
} from 'react-helsinki-headless-cms/apollo';
import { useTranslation } from 'react-i18next';
import { useLocation, useMatch } from 'react-router-dom';

import styles from './header.module.scss';
import {
  OrganisationNodeFieldsFragment,
  useMyProfileQuery,
} from '../../../generated/graphql';
import { HEADER_MENU_NAME } from '../../../headless-cms/constants';
import { useCmsLanguageOptions } from '../../../headless-cms/hooks';
import { stripLocaleFromUri } from '../../../headless-cms/utils';
import { useAppDispatch } from '../../../hooks/useAppDispatch';
import { useAppSelector } from '../../../hooks/useAppSelector';
import useLocale from '../../../hooks/useLocale';
import useNavigate from '../../../hooks/useNavigate';
import { skipFalsyType } from '../../../utils/typescript.utils';
import updateLocaleParam from '../../../utils/updateLocaleParam';
import useLogout from '../../auth/useLogout';
import { setActiveOrganisation } from '../../organisation/actions';
import { activeOrganisationSelector } from '../../organisation/selector';
import AppConfig from '../AppConfig';
import { ROUTES } from '../routes/constants';
import { getCmsPath } from '../routes/utils';

const UserNavigation: React.FC<{
  organisations: OrganisationNodeFieldsFragment[];
  userLabel: string;
}> = ({ organisations, userLabel }) => {
  const logout = useLogout();
  const { t } = useTranslation();
  const { pushWithLocale } = useNavigate();
  const dispatch = useAppDispatch();

  const goToEditMyProfile = () => {
    pushWithLocale(ROUTES.MY_PROFILE);
  };

  const goToHelsinkiProfile = () => {
    if (typeof window !== 'undefined') {
      window.open(
        AppConfig.helsinkiProfileUrl,
        '_blank',
        'noopener,noreferrer'
      );
    }
  };

  const activeOrganisation = useAppSelector(activeOrganisationSelector);

  const changeActiveOrganisation = (id: string) => {
    dispatch(setActiveOrganisation(id));
  };

  return (
    <HDSHeader.ActionBarItem
      id="user-menu"
      dropdownClassName={styles.userMenuDropdown}
      icon={<IconUser />}
      ariaLabel={t('header.userMenu.ariaLabelButton')}
      label={userLabel}
      preventButtonResize
    >
      <HDSButton
        key={`edit-profile`}
        className={styles.dropdownButton}
        iconLeft={<IconUser />}
        onClick={goToEditMyProfile}
        variant="supplementary"
      >
        {t('header.userMenu.openMyProfile')}
      </HDSButton>
      <HDSButton
        key={`go-to-profile`}
        className={styles.dropdownButton}
        iconLeft={<IconLinkExternal />}
        onClick={goToHelsinkiProfile}
        variant="supplementary"
      >
        {t('header.userMenu.openHelsinkiProfile')}
      </HDSButton>
      {organisations?.map((organisation) => (
        <HDSButton
          key={`organisation-${organisation.id}`}
          className={
            activeOrganisation === organisation.id
              ? styles.activeDropdownButton
              : styles.dropdownButton
          }
          iconLeft={<IconAngleRight />}
          onClick={() => changeActiveOrganisation(organisation.id)}
          variant="supplementary"
        >
          {organisation.name || ''}
        </HDSButton>
      ))}
      <HDSButton
        key={`sign-out`}
        className={styles.dropdownButton}
        iconLeft={<IconSignout aria-hidden />}
        onClick={logout}
        variant="supplementary"
      >
        {t('header.userMenu.logout')}
      </HDSButton>
    </HDSHeader.ActionBarItem>
  );
};

const Header: React.FC = () => {
  const { isAuthenticated } = useOidcClient();
  const { pushWithLocale } = useNavigate();
  const locale = useLocale();
  const isCmsPage = !!useMatch(`/${locale}${ROUTES.CMS_PAGE}`);
  const cmsLanguageOptions = useCmsLanguageOptions({ skip: !isCmsPage });
  const { pathname, search } = useLocation();

  const isLoggedIn = isAuthenticated();

  const getCmsHref = (language: LanguageCodeEnum) => {
    const nav = cmsLanguageOptions?.find((cmsLanguageOption) => {
      return cmsLanguageOption.locale?.toLowerCase() === language.toLowerCase();
    });

    return `/${language.toLowerCase()}${getCmsPath(
      stripLocaleFromUri(nav?.uri ?? '')
    )}`;
  };

  const getOriginHref = (language: LanguageCodeEnum): string => {
    const url = new URL(AppConfig.origin);
    url.pathname = updateLocaleParam(pathname, locale, language);
    url.search = search;
    return url.toString();
  };

  const getIsItemActive = (menuItem: MenuItem): boolean => {
    const pathWithoutTrailingSlash = (menuItem.path ?? '').replace(/\/$/, '');
    return (
      typeof window !== 'undefined' &&
      window.location.pathname.includes(
        `/${locale}${getCmsPath(stripLocaleFromUri(pathWithoutTrailingSlash))}`
      )
    );
  };

  const getPathnameForLanguage = (language: RHHCLanguage): string => {
    const languageCode = language.code ?? LanguageCodeEnum.Fi;
    return isCmsPage ? getCmsHref(languageCode) : getOriginHref(languageCode);
  };

  const goToPage =
    (pathname: string) =>
    (event?: React.MouseEvent<HTMLAnchorElement> | Event) => {
      event?.preventDefault();
      pushWithLocale(pathname);
    };

  const { data: myProfileData } = useMyProfileQuery({ skip: !isLoggedIn });

  const organisations: OrganisationNodeFieldsFragment[] =
    myProfileData?.myProfile?.organisations?.edges?.map((edge) => ({
      ...(edge?.node as OrganisationNodeFieldsFragment),
    })) ?? [];

  const languagesQuery = useLanguagesQuery();
  const languages = languagesQuery.data?.languages?.filter(
    skipFalsyType<RHHCLanguage | null>
  );

  const menuQuery = useMenuQuery({
    skip: !isLoggedIn,
    variables: { id: HEADER_MENU_NAME[locale], menuIdentifiersOnly: true },
  });
  const menu = menuQuery.data?.menu;

  return (
    <Navigation
      languages={languages}
      menu={menu}
      onTitleClick={goToPage(ROUTES.HOME)}
      getIsItemActive={getIsItemActive}
      getPathnameForLanguage={getPathnameForLanguage}
      userNavigation={
        isLoggedIn && (
          <UserNavigation
            organisations={organisations}
            userLabel={myProfileData?.myProfile?.name ?? ''}
          />
        )
      }
    />
  );
};

export default Header;
