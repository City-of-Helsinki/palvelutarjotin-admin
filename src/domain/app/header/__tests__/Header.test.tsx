import { graphql } from 'msw';
import * as React from 'react';

import { MyProfileDocument } from '../../../../generated/graphql';
import { initCmsMenuItemsMocks } from '../../../../test/cmsMocks';
import { server } from '../../../../test/msw/server';
import { fakePage } from '../../../../utils/cmsMockDataUtils';
import { fakePerson } from '../../../../utils/mockDataUtils';
import {
  render,
  screen,
  userEvent,
  waitFor,
} from '../../../../utils/testUtils';
import * as selectors from '../../../auth/selectors';
import Header from '../Header';

const profileResponse = {
  data: {
    myProfile: fakePerson(),
  },
};

const mocks = [
  {
    request: {
      query: MyProfileDocument,
    },
    result: profileResponse,
  },
];

beforeEach(() => {
  initCmsMenuItemsMocks();
  server.use(
    graphql.query('Page', (req, res, ctx) => {
      return res(
        ctx.data({
          page: fakePage(),
        })
      );
    })
  );
});

it('Header matches snapshot', () => {
  const { container } = render(<Header />, { mocks });
  expect(container.firstChild).toMatchSnapshot();
});

it('focuses skip link first', () => {
  render(<Header />);
  userEvent.tab();
  expect(screen.getByText('Siirry sisältöön')).toHaveFocus();
});

describe('cimode in language selector', () => {
  const OLD_ENV = process.env;

  beforeEach(() => {
    jest.resetModules(); // Most important - it clears the cache
    process.env = { ...OLD_ENV }; // Make a copy
  });

  afterAll(() => {
    process.env = OLD_ENV; // Restore old environment
  });

  test('cimode is rendered in a list of languages when feature flag is on', async () => {
    process.env.REACT_APP_LANGUAGE_CIMODE_VISIBLE = 'true';
    render(<Header />);
    await screen.findAllByText(/CIMODE/);
  });

  test('cimode is not rendered in a list of languages when feature flag is off', async () => {
    process.env.REACT_APP_LANGUAGE_CIMODE_VISIBLE = 'false';
    render(<Header />);
    expect(screen.queryAllByText(/CIMODE/).length).toBe(0);
  });
});

test('header renders cms menu items', async () => {
  jest.spyOn(selectors, 'isAuthenticatedSelector').mockReturnValue(true);
  const { menuItems } = initCmsMenuItemsMocks();
  render(<Header />, { mocks });
  await waitFor(() => {
    expect(
      screen.getByRole('button', {
        name: /fi kielivalikko/i,
      })
    ).toBeInTheDocument();
  });
  for (const menuItem of menuItems) {
    if (menuItem.children) {
      const dropdownButton = await screen.findByRole('button', {
        name: menuItem.title,
      });
      userEvent.click(dropdownButton);
      for (const childItem of menuItem.children) {
        await screen.findByRole('link', {
          name: childItem.title,
          hidden: true,
        });
      }
    } else {
      const link = await screen.findByRole('link', { name: menuItem.title });
      // eslint-disable-next-line jest/no-conditional-expect
      expect(link).toHaveAttribute('href', `/fi/cms-page/${menuItem.slug}`);
    }
  }
});
