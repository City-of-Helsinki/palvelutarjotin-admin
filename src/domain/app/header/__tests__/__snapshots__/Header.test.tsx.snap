// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header matches snapshot 1`] = `
<header
  class="hds-header Header-module_header__291GF"
  style="--header-max-width: var(--breakpoint-xl);"
>
  <div
    class="Header-module_headerBackgroundWrapper__KsFy2"
  >
    <a
      class="SkipLink-module_skipLink__1knzF"
      href="#main-content"
    >
      <span
        class="SkipLink-module_skipLinkLabel__21n5B"
      >
        Siirry sisältöön
      </span>
    </a>
    <div
      class="HeaderActionBar-module_headerActionBarContainer__5tfj-"
    >
      <div
        class="HeaderActionBar-module_headerActionBar__lPeGE"
      >
        <a
          aria-label="Helsingin kaupunki"
          class="HeaderActionBar-module_titleAndLogoContainer__2LEAb HeaderActionBar-module_logo__2r2T0"
          href="#"
          role="link"
        >
          <span
            class="HeaderActionBarLogo-module_logoWrapper__8NgfS"
          >
            <img
              alt="Helsinki"
              class="Logo-module_logo__Y2mwP Logo-module_large__3Tjj3"
              src="mocked hds-react logoFi"
            />
          </span>
        </a>
        <a
          class="HeaderActionBar-module_titleAndLogoContainer__2LEAb HeaderActionBar-module_title__opzt9 HeaderActionBar-module_normal__3B4qi"
          href="#"
        >
          <span
            class="HeaderActionBar-module_title__opzt9"
          >
            Kultus Admin
          </span>
        </a>
        <div
          class="HeaderActionBar-module_headerActions__3lNnM"
        >
          <div
            class="HeaderLanguageSelector-module_languageSelector__3d38B"
          >
            <div
              class="HeaderLanguageSelector-module_languageNodes__lTe8p"
            >
              <button
                aria-current="true"
                class="HeaderLanguageSelector-module_item__3_I7N HeaderLanguageSelector-module_activeItem__15LKg"
                lang="fi"
                type="button"
              >
                <span>
                  Suomi
                </span>
              </button>
              <button
                aria-current="false"
                class="HeaderLanguageSelector-module_item__3_I7N"
                lang="sv"
                type="button"
              >
                <span>
                  Svenska
                </span>
              </button>
              <button
                aria-current="false"
                class="HeaderLanguageSelector-module_item__3_I7N"
                lang="en"
                type="button"
              >
                <span>
                  English
                </span>
              </button>
            </div>
          </div>
          <div
            class="HeaderActionBarItem-module_container__1uq2m HeaderActionBarItem-module_hasContent__YJfOU HeaderActionBarItem-module_hasSubItems__1LVkh"
            id="user-menu"
          >
            <button
              aria-controls="user-menu-dropdown"
              aria-expanded="false"
              aria-label="Avaa käyttäjän valikko"
              class="HeaderActionBarItemButton-module_actionBarItemButton__2ALyH HeaderActionBarItem-module_icon__1Ug_k HeaderActionBarItemButton-module_hasSubItems__3wX3Y"
              type="button"
            >
              <div
                class="HeaderActionBarItemButton-module_buttonContent__3JeY0"
              >
                <span>
                  <span
                    class="HeaderActionBarItemButton-module_actionBarItemButtonIcon__1_1Cg"
                  >
                    <svg
                      aria-hidden="true"
                      aria-label="user"
                      class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                      role="img"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M16.2667 13C18.8082 13 20.8819 15.0031 20.9951 17.5167L21 17.7333V22H3V17.7333C3 15.1918 5.00309 13.1181 7.51667 13.0049L7.73333 13H16.2667ZM7.73333 15C6.27966 15 5.09104 16.1348 5.00499 17.5668L5 17.7333V20H19V17.7333C19 16.3172 17.8926 15.0973 16.54 15.0075L16.3766 15.0022L7.73333 15ZM12 2C14.7614 2 17 4.23858 17 7C17 9.76142 14.7614 12 12 12C9.23858 12 7 9.76142 7 7C7 4.23858 9.23858 2 12 2ZM12 4C10.3431 4 9 5.34315 9 7C9 8.65685 10.3431 10 12 10C13.6569 10 15 8.65685 15 7C15 5.34315 13.6569 4 12 4Z"
                        fill="currentColor"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                  <span
                    class="HeaderActionBarItemButton-module_actionBarItemButtonLabel__12Iec"
                  >
                    John Doe
                  </span>
                </span>
                <svg
                  aria-hidden="true"
                  aria-label="angle-down"
                  class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                  role="img"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M12 13.5L17 8.5L18.5 10L12 16.5L5.5 10L7 8.5L12 13.5Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </button>
            <div
              class="HeaderActionBarItem-module_dropdownWrapper__39cBw"
            >
              <div
                class="HeaderActionBarItem-module_dropdown__3daDZ userMenuDropdown HeaderActionBarItem-module_hasContent__YJfOU HeaderActionBarItem-module_hasSubItems__1LVkh"
                id="user-menu-dropdown"
              >
                <ul>
                  <button
                    class="Button-module_button__1msFE button_hds-button__2A0je Button-module_supplementary__3YKiS button_hds-button--supplementary__GcHcV dropdownButton"
                    type="button"
                  >
                    <div
                      aria-hidden="true"
                      class="Button-module_icon__O-h7R button_hds-icon__17j8Z"
                    >
                      <svg
                        aria-hidden="true"
                        aria-label="user"
                        class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                        role="img"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M16.2667 13C18.8082 13 20.8819 15.0031 20.9951 17.5167L21 17.7333V22H3V17.7333C3 15.1918 5.00309 13.1181 7.51667 13.0049L7.73333 13H16.2667ZM7.73333 15C6.27966 15 5.09104 16.1348 5.00499 17.5668L5 17.7333V20H19V17.7333C19 16.3172 17.8926 15.0973 16.54 15.0075L16.3766 15.0022L7.73333 15ZM12 2C14.7614 2 17 4.23858 17 7C17 9.76142 14.7614 12 12 12C9.23858 12 7 9.76142 7 7C7 4.23858 9.23858 2 12 2ZM12 4C10.3431 4 9 5.34315 9 7C9 8.65685 10.3431 10 12 10C13.6569 10 15 8.65685 15 7C15 5.34315 13.6569 4 12 4Z"
                          fill="currentColor"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </div>
                    <span
                      class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
                    >
                      Omat tiedot
                    </span>
                  </button>
                  <button
                    class="Button-module_button__1msFE button_hds-button__2A0je Button-module_supplementary__3YKiS button_hds-button--supplementary__GcHcV dropdownButton"
                    type="button"
                  >
                    <div
                      aria-hidden="true"
                      class="Button-module_icon__O-h7R button_hds-icon__17j8Z"
                    >
                      <svg
                        aria-hidden="true"
                        aria-label="link-external"
                        class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                        role="img"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M5 3V19H21V21H3V3H5ZM21 3V12H19V6.413L9.91421 15.5L8.5 14.0858L17.585 5H12V3H21Z"
                          fill="currentColor"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </div>
                    <span
                      class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
                    >
                      Helsinki Profiili
                    </span>
                  </button>
                  <button
                    class="Button-module_button__1msFE button_hds-button__2A0je Button-module_supplementary__3YKiS button_hds-button--supplementary__GcHcV dropdownButton"
                    type="button"
                  >
                    <div
                      aria-hidden="true"
                      class="Button-module_icon__O-h7R button_hds-icon__17j8Z"
                    >
                      <svg
                        aria-hidden="true"
                        aria-label="signout"
                        class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                        role="img"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M16 2V7H14V4H5V20H14V17H16V22H3V2H16ZM18 7L23 12L18 17L16.5 15.5L19 12.999L8 13V11L18.999 10.999L16.5 8.5L18 7Z"
                          fill="currentColor"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </div>
                    <span
                      class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
                    >
                      Kirjaudu ulos
                    </span>
                  </button>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="HeaderNavigationMenu-module_headerNavigationMenuContainer__2nK9z"
    >
      <nav
        class="HeaderNavigationMenu-module_headerNavigationMenu__3G6-N"
      >
        <ul
          class="HeaderNavigationMenu-module_headerNavigationMenuList__3HIzx"
        >
          <li>
            <span
              class="HeaderNavigationMenu-module_headerNavigationMenuLinkContent__3zC2j"
            >
              <span
                class="HeaderLink-module_navigationLinkWrapper__2jd7A"
              >
                <a
                  class="HeaderLink-module_headerLink__2i9-J HeaderNavigationMenu-module_headerNavigationMenuLink__H_ogS"
                  href="/fi/cms-page/mika-kultus-2/"
                  id="cG9zdDo0MDk="
                >
                  Mikä kultus?
                </a>
                <button
                  aria-expanded="false"
                  aria-label="Avaa alasvetovalikko."
                  class="HeaderLinkDropdown-module_button__2xVuy HeaderLinkDropdown-module_depth-0__Xeyks"
                  data-testid="dropdown-button-0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    aria-label="angle-down"
                    class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik HeaderLinkDropdown-module_chevron__1KsTd"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 13.5L17 8.5L18.5 10L12 16.5L5.5 10L7 8.5L12 13.5Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </button>
                <ul
                  class="HeaderLinkDropdown-module_dropdownMenu__3oeWC"
                  data-testid="dropdown-menu-0"
                  style="display: none;"
                >
                  <li>
                    <span
                      class="HeaderLink-module_navigationLinkWrapper__2jd7A"
                    >
                      <a
                        class="HeaderLink-module_headerLink__2i9-J HeaderLinkDropdown-module_dropdownLink__2UIF3"
                        href="/fi/cms-page/mika-kultus-2/katrin-testisivu/"
                        id="cG9zdDo0MTE="
                      >
                        Katrin testisivu
                      </a>
                    </span>
                  </li>
                </ul>
              </span>
            </span>
          </li>
          <li>
            <span
              class="HeaderNavigationMenu-module_headerNavigationMenuLinkContent__3zC2j"
            >
              <span
                class="HeaderLink-module_navigationLinkWrapper__2jd7A"
              >
                <a
                  class="HeaderLink-module_headerLink__2i9-J HeaderNavigationMenu-module_headerNavigationMenuLink__H_ogS"
                  href="/fi/cms-page/kulttuurikasvatus/"
                  id="cG9zdDo0MTI="
                >
                  Kulttuurikasvatus
                </a>
                <button
                  aria-expanded="false"
                  aria-label="Avaa alasvetovalikko."
                  class="HeaderLinkDropdown-module_button__2xVuy HeaderLinkDropdown-module_depth-0__Xeyks"
                  data-testid="dropdown-button-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    aria-label="angle-down"
                    class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik HeaderLinkDropdown-module_chevron__1KsTd"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 13.5L17 8.5L18.5 10L12 16.5L5.5 10L7 8.5L12 13.5Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </button>
                <ul
                  class="HeaderLinkDropdown-module_dropdownMenu__3oeWC"
                  data-testid="dropdown-menu-1"
                  style="display: none;"
                >
                  <li>
                    <span
                      class="HeaderLink-module_navigationLinkWrapper__2jd7A"
                    >
                      <a
                        class="HeaderLink-module_headerLink__2i9-J HeaderLinkDropdown-module_dropdownLink__2UIF3"
                        href="/fi/cms-page/kulttuurikasvatus/kulttuuripolulla-kulttuuria-kaikille/"
                        id="cG9zdDo0MTQ="
                      >
                        Kulttuuripolulla kulttuuria kaikille
                      </a>
                    </span>
                  </li>
                  <li>
                    <span
                      class="HeaderLink-module_navigationLinkWrapper__2jd7A"
                    >
                      <a
                        class="HeaderLink-module_headerLink__2i9-J HeaderLinkDropdown-module_dropdownLink__2UIF3"
                        href="/fi/cms-page/kulttuurikasvatus/kulttuurikoordinaattori-olet-meille-tarkea/"
                        id="cG9zdDo0MTU="
                      >
                        Kulttuurikoordinaattori, olet meille tärkeä!
                      </a>
                    </span>
                  </li>
                  <li>
                    <span
                      class="HeaderLink-module_navigationLinkWrapper__2jd7A"
                    >
                      <a
                        class="HeaderLink-module_headerLink__2i9-J HeaderLinkDropdown-module_dropdownLink__2UIF3"
                        href="/fi/cms-page/kulttuurikasvatus/kulttuurikasvatuksen-kasitteita/"
                        id="cG9zdDo0MTY="
                      >
                        Kulttuurikasvatuksen käsitteitä
                      </a>
                    </span>
                  </li>
                </ul>
              </span>
            </span>
          </li>
          <li>
            <span
              class="HeaderNavigationMenu-module_headerNavigationMenuLinkContent__3zC2j"
            >
              <span
                class="HeaderLink-module_navigationLinkWrapper__2jd7A"
              >
                <a
                  class="HeaderLink-module_headerLink__2i9-J HeaderNavigationMenu-module_headerNavigationMenuLink__H_ogS"
                  href="/fi/cms-page/oppimateriaalit-2/"
                  id="cG9zdDo0MTc="
                >
                  Oppimateriaalit
                </a>
                <button
                  aria-expanded="false"
                  aria-label="Avaa alasvetovalikko."
                  class="HeaderLinkDropdown-module_button__2xVuy HeaderLinkDropdown-module_depth-0__Xeyks"
                  data-testid="dropdown-button-2"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    aria-label="angle-down"
                    class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik HeaderLinkDropdown-module_chevron__1KsTd"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 13.5L17 8.5L18.5 10L12 16.5L5.5 10L7 8.5L12 13.5Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </button>
                <ul
                  class="HeaderLinkDropdown-module_dropdownMenu__3oeWC"
                  data-testid="dropdown-menu-2"
                  style="display: none;"
                >
                  <li>
                    <span
                      class="HeaderLink-module_navigationLinkWrapper__2jd7A"
                    >
                      <a
                        class="HeaderLink-module_headerLink__2i9-J HeaderLinkDropdown-module_dropdownLink__2UIF3"
                        href="/fi/cms-page/oppimateriaalit-2/ylakoulu-ja-toinen-aste/"
                        id="cG9zdDo0MTk="
                      >
                        Yläkoulu ja toinen aste
                      </a>
                    </span>
                  </li>
                  <li>
                    <span
                      class="HeaderLink-module_navigationLinkWrapper__2jd7A"
                    >
                      <a
                        class="HeaderLink-module_headerLink__2i9-J HeaderLinkDropdown-module_dropdownLink__2UIF3"
                        href="/fi/cms-page/oppimateriaalit-2/varhaiskasvatus-ja-esiopetus/"
                        id="cG9zdDo0MjA="
                      >
                        Varhaiskasvatus ja esiopetus
                      </a>
                    </span>
                  </li>
                  <li>
                    <span
                      class="HeaderLink-module_navigationLinkWrapper__2jd7A"
                    >
                      <a
                        class="HeaderLink-module_headerLink__2i9-J HeaderLinkDropdown-module_dropdownLink__2UIF3"
                        href="/fi/cms-page/oppimateriaalit-2/alakoulu/"
                        id="cG9zdDo0MjE="
                      >
                        Alakoulu
                      </a>
                    </span>
                  </li>
                </ul>
              </span>
            </span>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
